src/exec.c

```diff
@@ -506,6 +500,18 @@ static int hyper_do_exec_cmd(struct hyper_exec *exec, int pipe, struct stdio_con
 	else
 		unsetenv("TERM");
 
+	if (exec->init == 0) {
```

Comment:
exec env and workdir are processed in hyper_exec_process() right after this. Do you mean to limit it to non-init exec case?